ppoprf/src/ppoprf.rs

```diff
@@ -87,6 +135,115 @@ impl ProofDLEQ {
     c_prime == self.c
   }
 
+  fn verify_batch(
+    &self,
+    public_value: &RistrettoPoint,
+    p: &[RistrettoPoint], //P
+    q: &[RistrettoPoint], //Q
+  ) -> bool {
+    //(M, Z) = ComputeComposites(B, C, D)
+    let (m, z) = ProofDLEQ::compute_composites(None, public_value, p, q);
+
+    //t2 = ((s * A) + (c * B))
+    let t2 = (self.s * RISTRETTO_BASEPOINT_POINT) + (self.c * public_value);
+    //t3 = ((s * M) + (c * Z))
+    let t3 = (self.s * m) + (self.c * z);
+
+    /*challengeTranscript =
+    I2OSP(len(Bm), 2) || Bm ||
+    I2OSP(len(a0), 2) || a0 ||
+    I2OSP(len(a1), 2) || a1 ||
+    I2OSP(len(a2), 2) || a2 ||
+    I2OSP(len(a3), 2) || a3 ||
+    "Challenge"*/
+    let mut challenge_transcript = Vec::new();
+    let compressed_point_len_slice = &ProofDLEQ::i2osp2(COMPRESSED_POINT_LEN);
+    challenge_transcript.extend_from_slice(compressed_point_len_slice); //len(Bm)
+    challenge_transcript.extend_from_slice(public_value.compress().as_bytes()); //Bm
+    challenge_transcript.extend_from_slice(compressed_point_len_slice); //len(a0)
+    challenge_transcript.extend_from_slice(m.compress().as_bytes()); //a0 = m
+    challenge_transcript.extend_from_slice(compressed_point_len_slice); //len(a1)
+    challenge_transcript.extend_from_slice(z.compress().as_bytes()); //a1 = z
+    challenge_transcript.extend_from_slice(compressed_point_len_slice); //len(a2)
+    challenge_transcript.extend_from_slice(t2.compress().as_bytes()); //a2 = t2
+    challenge_transcript.extend_from_slice(compressed_point_len_slice); //len(a3)
+    challenge_transcript.extend_from_slice(t3.compress().as_bytes()); //a3 = t3
+
+    //c = G.HashToScalar(challengeTranscript)
+    let c = ProofDLEQ::hash_to_scalar(&challenge_transcript, "Challenge");
```

Comment:
here we can follow how it is done in other parts of the code: https://github.com/brave/sta-rs/blob/main/ppoprf/src/ppoprf.rs#L59

Reply:
I removed the `hash` function, as it was used nowhere else in the code.